packages/runner-shared/src/logger.js

```diff
@@ -30,6 +30,13 @@ export const logger = {
     _.each(formattedLog, (value, key) => {
       // don't log empty strings
       // _.trim([]) returns '' but we want to log empty arrays, so account for that
+      // If value is a javascript object that isn't an array or null we need to stringify object.
+      if (value
+        && typeof value === 'object'
+        && !_.isArray(value)) {
+        value = JSON.stringify(value)
+      }
+
       if (_.trim(value) === '' && !_.isArray(value)) return
```

Comment:
The error gets thrown on this `_.trim` call, so wouldn't it make more sense to just skip the `_.trim` call if we know it's an object?

Something like this:

```suggestion
      if (typeof value !== 'object' && _.trim(value) === '' && !_.isArray(value)) return
```